twake/frontend/src/app/components/Leftbar/Channel/ChannelMenu.tsx

```diff
@@ -0,0 +1,138 @@
+  channel: ChannelType;
+};
+
+export default (props: Prpos): JSX.Element => {
+  const userName = UserService.getCurrentUser().username;
+  Languages.useListener(useState);
+  Notifications.useListener(useState);
+  //@ts-ignore
+  const hasNotification = (Notifications.notification_by_channel[props.channel.id] || {}).count > 0;
+
+  const addOrCancelFavorite = () => {};
+
+  const displayMembers = () => {};
+
+  const leaveChannel = () => {};
+
+  const editChannel = () => {
+    ModalManager.open(
+      <ChannelWorkspaceEditor title={'Edit Channel'} channel={props.channel || {}} />,
+      {
+        position: 'center',
+        size: { width: '600px' },
+      },
+    );
+  };
+
+  const removeChannel = () => {};
+
+  // Add translations
+  let menu: object[] = [
+    {
+      type: 'menu',
+      text: Languages.t('scenes.apps.messages.left_bar.stream.notifications'),
+      onClick: () => {},
+      submenu: [
+        {
+          text: Languages.t('scenes.apps.messages.left_bar.stream.notifications.all'),
+          icon: 'check',
+        },
+        {
+          text: Languages.t('scenes.apps.messages.left_bar.stream.notifications.mentions', [
+            '@all',
+            '@here',
+            `@${userName}`,
+          ]),
+        },
+        {
+          text: Languages.t('scenes.apps.messages.left_bar.stream.notifications.me', [
+            `@${userName}`,
+          ]),
+        },
+        { text: Languages.t('scenes.apps.messages.left_bar.stream.notifications.never') },
+      ],
+    },
+    {
+      type: 'menu',
+      text: Languages.t(
+        hasNotification ? 'scenes.app.channelsbar.read_sign' : 'scenes.app.channelsbar.unread_sign',
+      ),
+      onClick: () => {
+        hasNotification ? Notifications.read(props.channel) : Notifications.unread(props.channel);
+      },
+    },
+    {
+      type: 'menu',
+      text: Languages.t(
+        props.channel.user_member?.favorite
+          ? 'scenes.apps.messages.left_bar.stream.add_to_favorites'
+          : 'scenes.apps.messages.left_bar.stream.remove_from_favorites',
+      ),
+      onClick: () => {
+        addOrCancelFavorite();
+      },
+    },
+    {
+      type: 'separator',
+    },
+    {
+      type: 'menu',
+      text: Languages.t('scenes.apps.parameters.workspace_sections.members'),
+      onClick: () => {
+        displayMembers();
+      },
+    },
+    {
+      type: 'menu',
+      text: Languages.t('scenes.app.channelsbar.modify_channel_menu'),
+      onClick: () => {
+        editChannel();
+      },
+    },
+    {
+      type: 'menu',
+      text: Languages.t('scenes.app.channelsbar.channel_leaving'),
+      className: 'menu-text-delicate',
+      onClick: () => {
+        AlertManager.confirm(() => leaveChannel());
+      },
+    },
+  ];
+
+  //To do: Add the necessery admin rights
+  if (true) {
+    menu.push({
+      type: 'menu',
+      text: Languages.t('scenes.app.channelsbar.channel_removing'),
+      className: 'menu-text-delicate',
```

Comment:
We already have a danger class for menu items: "danger", please use this one :)

Reply:
Done